crates/ruma-common/src/events/room.rs

```diff
@@ -106,6 +123,45 @@ impl ImageInfo {
     pub fn new() -> Self {
         Self::default()
     }
+
+    /// Create an `ImageInfo` from the given file info, image info and thumbnail.
+    #[cfg(feature = "unstable-msc3552")]
+    pub fn from_extensible_content(
+        file_info: Option<&FileContentInfo>,
+        image: &ImageContent,
+        thumbnail: &[ThumbnailContent],
+    ) -> Option<Self> {
+        if file_info.is_none() && image.is_empty() && thumbnail.is_empty() {
+            None
+        } else {
+            let (mimetype, size) = file_info
+                .map(|info| (info.mimetype.to_owned(), info.size.to_owned()))
+                .unwrap_or_default();
+            let ImageContent { height, width } = image.to_owned();
+            let (thumbnail_source, thumbnail_info) = thumbnail
+                .get(0)
+                .map(|thumbnail| {
+                    let source = (&thumbnail.file).into();
+                    let info = ThumbnailInfo::from_extensible_content(
+                        thumbnail.file.info.as_deref(),
+                        thumbnail.image.as_deref(),
+                    )
+                    .map(Box::new);
+                    (Some(source), info)
+                })
+                .unwrap_or_default();
+
+            Some(Self {
+                height,
+                width,
+                mimetype,
+                size,
+                thumbnail_source,
+                thumbnail_info,
+                blurhash: None,
```

Comment:
MSC3552 doesn't say anything about blurhashes. Seems to me like this line should have an extra `#[cfg(feature = "…")]` above for the blurhash MSC.

Reply:
If it's unclear, we can leave blurhash support for the new events out entirely.